src/analyses/loop_termination.ml

```diff
@@ -0,0 +1,116 @@
+(** Work in progress *)
+
+open Analyses
+open GoblintCil
+open TerminationPreprocessing
+
+exception PreProcessing of string
+
+(*
+let loop_heads () =
+  let module FileCfg =
+  struct
+    let file = !Cilfacade.current_file
+    module Cfg = (val !MyCFG.current_cfg)
+  end in
+  let module WitnessInvariant = WitnessUtil.Invariant (FileCfg) in
+  WitnessInvariant.loop_heads (* TODO: Unused *)
+*)
+
+(** Contains all loop counter variables (varinfo) and maps them to their corresponding loop statement. *)
+let loop_counters : stmt VarToStmt.t ref = ref VarToStmt.empty
+
+(** Contains the locations of the upjumping gotos *)
+let upjumping_gotos : location list ref = ref []
+
+(** Indicates the place in the code, right after a loop is exited. *)
+let loop_exit : varinfo ref = ref (makeVarinfo false "-error" Cil.intType)
+
+let is_loop_counter_var (x : varinfo) =
+  VarToStmt.mem x !loop_counters
+
+let is_loop_exit_indicator (x : varinfo) =
+  x = !loop_exit
```

Comment:
Iirc, CIL `varinfo` should not be compared with physical equality, as this may fail for the `type`s that are in it.